code/game/objects/items/devices/flashlight.dm

```diff
@@ -59,8 +59,12 @@
 		to_chat(user, SPAN_WARNING("You cannot toggle \the [src.name] on or off."))
 		return FALSE
 	if(!isturf(user.loc))
-		to_chat(user, "You cannot turn the light on while in [user.loc].") //To prevent some lighting anomalities.
-		return FALSE
+		if(on)
+			to_chat(user, SPAN_WARNING("You cannot turn the light off while in [user.loc].")) //To prevent some lighting anomalities.
```

Comment:
instead of the if and else, couldn't you just make it use the "on" variable? Something like `"You cannot turn the light [on ? "off" : "on"] while in [user.loc]."`.

Reply:
if(!isturf(user.loc))
		to_chat(user, SPAN_WARNING("You cannot turn the light off or on while in [user.loc].")) //To prevent some lighting anomalities.

i replace my code with this seem more simpler and work.